lib/krikri/random_record_generator.rb

```diff
@@ -1,19 +1,36 @@
 module Krikri
   # Gets random records from the search index
   class RandomRecordGenerator
+    include Krikri::QaProviderFilter
 
     def initialize
-      @solr_repo = Blacklight::SolrRepository.new(Blacklight::Configuration.new)
+      @index_querier = Krikri::IndexQuerier.new
+      @default_query_params = { :id => '*:*',
```

Comment:
The search matches all of a provider's documents, sorted randomly, and returns only one document.